src/Microsoft.AspNetCore.Routing/Builder/EndpointDataSourceBuilderExtensions.cs

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Routing.Patterns;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    public static class EndpointDataSourceBuilderExtensions
+    {
+        #region Http
+        public static IApplyEndpointBuilder MapGet(
```

Comment:
What if we did the following:
```
EndpointBuilder -> EndpointModel
IApplyEndpointBuilder -> IEndpointModel
```

Reply:
Hmm, I would expect EndpointModel to implement IEndpointModel, and the members to be the same.